include/adm/serial/frame_format.hpp

```diff
@@ -0,0 +1,228 @@
+   * | numMetadataChunks      | :type:`NumMetadataChunks`      | :class:`OptionalParameter`   |
+   * +------------------------+--------------------------------+------------------------------+
+   * | countToSameChunk       | :type:`CountToSameChunk`       | :class:`OptionalParameter`   |
+   * +------------------------+--------------------------------+------------------------------+
+   * | changedIDs             | :type:`ChangedIds`             | :class:`OptionalParameter`   |
+   * +------------------------+--------------------------------+------------------------------+
+   * \endrst
+   *
+   * Note that:
+   * - countToFull has defaults dependent on the flow type, as this is not known at this level they are not implemented
+   * - numMetadataChunks is required for frame type divided and not permitted for other frame types.
+   * - countToSameChunk is optional for frame type divided and not permitted for other frame types.
+   */
+  class FrameFormat : private detail::FrameFormatBase {
+   public:
+    using tag = FrameFormatTag;
+    /// Type that holds the id for this element;
+    using id_type = FrameFormatId;
+
+    /**
+     * @brief Constructor template
+     *
+     * Templated constructor which accepts a variable number of ADM parameters
+     * in random order after the mandatory ADM parameters.
+     */
+    template <typename... Parameters>
+    FrameFormat(FrameFormatId id, Start start, Duration duration,
+                FrameType frameType, Parameters... optionalNamedArgs);
+    ADM_EXPORT FrameFormat(const FrameFormat &) = default;
+    ADM_EXPORT FrameFormat(FrameFormat &&) = default;
+    ADM_EXPORT FrameFormat &operator=(const FrameFormat &) = default;
+    ADM_EXPORT FrameFormat &operator=(FrameFormat &&) = default;
+
+    /**
+     * @brief ADM parameter getter template
+     *
+     * Templated getter with the wanted ADM parameter type as template
+     * argument. If currently no value is available trying to get the adm
+     * parameter will result in an exception. Check with the has method before
+     */
+    template <typename Parameter>
+    Parameter get() const;
+
+    /**
+     * @brief ADM parameter has template
+     *
+     * Templated has method with the ADM parameter type as template argument.
+     * Returns true if the ADM parameter is set or has a default value.
+     */
+    template <typename Parameter>
+    bool has() const;
+
+    /**
+     * @brief ADM parameter isDefault template
+     *
+     * Templated isDefault method with the ADM parameter type as template
+     * argument. Returns true if the ADM parameter is the default value.
+     */
+    template <typename Parameter>
+    bool isDefault() const;
+
+    using detail::FrameFormatBase::get;
+    using detail::FrameFormatBase::has;
+    using detail::FrameFormatBase::isDefault;
+    using detail::FrameFormatBase::set;
+    using detail::FrameFormatBase::unset;
```

Comment:
make private except for set

Reply:
e90eb85ce094b93d514f5366c969328ee263cb17